src/main/java/core/basesyntax/service/impl/SupplierServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.handler.SupplierHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+
+public class SupplierServiceImpl implements OperationHandler {
+    private final SupplierHandler supplierHandler;
+
+    public SupplierServiceImpl(SupplierHandler supplierHandler) {
+        this.supplierHandler = supplierHandler;
+    }
+
+    @Override
+    public boolean applyOperation(FruitTransaction fruitTransaction) {
+        FruitTransaction newFruit = new FruitTransaction.FruitBuilder()
+                .setOperationType(fruitTransaction.getOperationType())
+                .setFruitName(fruitTransaction.getFruitName())
+                .setFruitQuantity(fruitTransaction.getFruitQuantity())
+                .setOperationType(fruitTransaction.getOperationType())
+                .build();
+        return supplierHandler.add(newFruit);
```

Comment:
do you really need so many redundant class? 
Here is diagram that should help you to improve your architecture 
<img width="864" alt="image" src="https://github.com/mate-academy/jv-fruit-shop/assets/32339836/9b84a904-26e7-4740-8b61-358dc28a00ed">